provider/core/src/lib.rs

```diff
@@ -248,7 +248,23 @@ pub mod prelude {
     pub use crate::DynamicDataProvider;
     #[doc(no_inline)]
     pub use crate::KeyedDataMarker;
+    #[doc(no_inline)]
+    #[doc(no_inline)]
+    pub use icu_locid::extensions::unicode::{
+        key as unicode_extension_key, Key as UnicodeExtensionKey,
+    };
+    #[doc(no_inline)]
+    pub use icu_locid::extensions::unicode::{
+        value as unicode_extension_value, Value as UnicodeExtensionValue,
+    };
+    #[doc(no_inline)]
+    pub use icu_locid::{langid, LanguageIdentifier};
```

Comment:
Issue: I don't think we should re-export types from other crates without renaming them or namespacing them. (I realize this is somewhat in conflict with my previous comment about not inventing new names, but see below)